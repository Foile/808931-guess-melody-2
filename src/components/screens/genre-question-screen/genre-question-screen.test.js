import React from 'react';
import renderer from "react-test-renderer";
import GenreQuestionScreen from './genre-question-screen';

const questionMock = {
  "type": `genre`,
  "genre": `reggae`,
  "answers": [
    {
      "src": `https://es31-server.appspot.com/guess-melody/static/music/Addis_Ababa.mp3`,
      "genre": `reggae`
    },
    {
      "src": `https://es31-server.appspot.com/guess-melody/static/music/Azure.mp3`,
      "genre": `electronic`
    },
    {
      "src": `https://es31-server.appspot.com/guess-melody/static/music/Whaling_City.mp3`,
      "genre": `country`
    },
    {
      "src": `https://es31-server.appspot.com/guess-melody/static/music/Skanada.mp3`,
      "genre": `reggae`
    }
  ]
};

it(`GenreQuestionScreen correctly renders after relaunch`, () => {
  const tree = renderer
  .create(<GenreQuestionScreen
    question={questionMock}
    screenIndex={0}
    onAnswer={jest.fn()}
  />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
