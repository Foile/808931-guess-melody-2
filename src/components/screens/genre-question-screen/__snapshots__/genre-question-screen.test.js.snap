// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreQuestionScreen correctly renders after relaunch 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Выберите 
    reggae
     треки
  </h2>
  <form
    className="game__tracks"
    onSubmit={[Function]}
  >
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        disabled={true}
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="answer-0"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="reggae"
        />
        <label
          className="game__check"
          htmlFor="answer-0"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        disabled={true}
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="answer-1"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="electronic"
        />
        <label
          className="game__check"
          htmlFor="answer-1"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        disabled={true}
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="answer-2"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="country"
        />
        <label
          className="game__check"
          htmlFor="answer-2"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        disabled={true}
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="answer-3"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="reggae"
        />
        <label
          className="game__check"
          htmlFor="answer-3"
        >
          Отметить
        </label>
      </div>
    </div>
    <button
      className="game__submit button"
      type="submit"
    >
      Ответить
    </button>
  </form>
</section>
`;
